### ISSUE TITLE
Is there a way to set main workspace ?

### ISSUE THREAD

**Issue by fthuin** [google/blockly#953]
Hello all,

I'm struggling to make a web page with multiple blockly workspace. Blockly calls `Blockly.getMainWorkspace()` to get the dragMode or the toolbox, but it returns `null` in my case.

The doc says 

> Returns the main workspace. Returns the last used main workspace (based on focus). Try not to use this function, particularly if there are multiple Blockly instances on a page.

Is there a way to set the main workspace ? Or a working example with multiple blockly workspace instances ?

Thanks!

PS: I've tried to use `markFocused()` on the workspace but I didn't manage to solve my problem.

**Comment by NeilFraser** [comment]
`Blockly.inject` returns the newly created workspace.  You probably want to save a reference to that.  Here's an example:
https://developers.google.com/blockly/guides/configure/web/fixed-size

Also, questions such as these are best sent to [the newsgroup](https://groups.google.com/forum/#!forum/blockly), that way they can be answered faster and the knowledge is available for subsequent users.

**Comment by NeilFraser** [comment]
Also, the [mirror demo](https://blockly-demo.appspot.com/static/demos/mirror/index.html) is a good example of two Blockly instances on the same page.  As is the Block Factory.  The `markFocused()` function is indeed the function for changing the current workspace.

**Comment by fthuin** [comment]
Sorry, I will post there next time.

Just for information if someone faces the problem in the future: I was trying to move workspaces between div elements in my HTML using the jQuery .remove() method instead of the jQuery.detach(). The former one modifies somehow the injectionDiv and made Blockly lose the link to the main workspace.
